src/core/LfoController.cpp

```diff
@@ -238,9 +239,13 @@ void LfoController::loadSettings( const QDomElement & _this )
 	m_waveModel.loadSettings( _this, "wave" );
 	m_multiplierModel.loadSettings( _this, "multiplier" );
 
-	if (!_this.attribute("userwavefile").isEmpty())
+	if (const auto userWaveFile = _this.attribute("userwavefile"); !userWaveFile.isEmpty())
 	{
-		m_userDefSampleBuffer = gui::SampleLoader::createBufferFromFile(_this.attribute("userwavefile"), true);
+		if (QFileInfo(PathUtil::toAbsolute(userWaveFile)).exists())
+		{
+			m_userDefSampleBuffer = gui::SampleLoader::createBufferFromFile(_this.attribute("userwavefile"));
+		}
+		else { Engine::getSong()->collectError(QString("%1: %2").arg(tr("Sample not found"), userWaveFile)); }
```

Comment:
Now that you've removed the error reporting from `SampleLoader::createBufferFromFile` and moved it here, do you need to construct a blank `SampleBuffer` to assign to `m_userDefSampleBuffer` like you used to do in `SampleLoader::createBufferFromFile`? Or is it okay to leave it with a nullptr?

Same question applies for other places that called `SampleLoader::createBufferFromFile`.

Reply:
On the current master branch, `nullptr` was not used to signify an empty buffer, but rather a buffer with one empty frame did that if there was a failure to load one in, so they should be default constructed to avoid surprises where some part of the code expects a valid buffer.

Reply:
Yes, I was trying to say that with your recent commit, it looks like you're accidentally no longer default constructing one when a sample fails to load.

Reply:
Or maybe I misread it? Let me see..

Reply:
> Yes, I was trying to say that with your recent commit, it looks like you're accidentally no longer default constructing one when a sample fails to load.

Wasn't by accident, lol, I knew about this and only made one default constructed in e44f6e16dfb890599125da57b185fb7e224abce8 ([EnvelopeAndLfoParameters.h](https://github.com/LMMS/lmms/commit/e44f6e16dfb890599125da57b185fb7e224abce8#diff-f124fd2feb341674be35db959b2a24463e7b6ef5cc9a78b15a27b3a789fd2c7b)), just wanted to get that commit out of the way first.